File: 1966-frequency-of-the-most-frequent-element/1966-frequency-of-the-most-frequent-element.py
class Solution:
    def maxFrequency(self, nums: List[int], k: int) -> int:
        nums.sort()
        left = 0
        summ = 0
        best = 0

        for right, n in enumerate(nums):
            summ += n

            while (right - left + 1) * n - summ > k:
                summ -= nums[left]
                left += 1
            
            best = max(right - left + 1, best)
        
        return best

File: 1966-frequency-of-the-most-frequent-element/1966-frequency-of-the-most-frequent-element.cpp
class Solution {
public:
    int maxFrequency(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());

        long windowSum = 0;
        int windowStart = 0;
        int best = 0;

        for (int i = 0; i < nums.size(); i++) {
            windowSum += nums[i];
            long required = long(i - windowStart + 1) * nums[i] - windowSum;
            while (required > k) {
                windowSum -= nums[windowStart];
                windowStart++;
                required = long(i - windowStart + 1) * nums[i] - windowSum;
            }
            best = max(best, i - windowStart + 1);
        }

        return best;
    }
};